File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutionsa/Skills/SkillDefinition.cs
using System.Collections.Generic;
using Microsoft.Bot.Configuration;
using Newtonsoft.Json;

namespace Microsoft.Bot.Solutions.Skills
{
    public class SkillDefinition : ConnectedService
    {
        public SkillDefinition()
            : base("skill")
        {
        }

        [JsonProperty("dispatchIntent")]
        public string DispatchIntent { get; set; }

        [JsonProperty("assembly")]
        public string Assembly { get; set; }

        [JsonProperty("luisServiceIds")]
        public string[] LuisServiceIds { get; set; }

        [JsonProperty("supportedProviders")]
        public string[] SupportedProviders { get; set; }

        [JsonProperty("parameters")]
        public string[] Parameters { get; set; }

        [JsonProperty("configuration")]
        public Dictionary<string, string> Configuration { get; set; } = new Dictionary<string, string>();

        public override void Decrypt(string secret)
        {
            base.Decrypt(secret);
        }

        public override void Encrypt(string secret)
        {
            base.Encrypt(secret);
        }
    }
}

File: solutions/Virtual-Assistant/src/typescript/bot-solution/src/skills/skillDefinition.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { ConnectedService, ServiceTypes } from 'botframework-config';

export class SkillDefinition extends ConnectedService {

    public dispatchIntent: string = '';

    public assembly: string = '';

    public luisServiceIds: string[] = [];

    public supportedProviders: string[] = [];

    public parameters: string[] = [];

    public configuration: Map<string, string> = new Map();

    constructor() {
        super(undefined, ServiceTypes.Generic);
    }
}
